File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    let result = []
    
    for (let i = 0; i < photo.length; i++) {
        let sum = 0
        for (let j = 0; j < photo[i].length; j++) {
            for (let k = 0; k < name.length; k++) {
                if (name[k] === photo[i][j]) {
                    sum += yearning[k]
                }
            }
        }
        result.push(sum)
    }
    return result
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    name_yearning_map = {n: y for n, y in zip(name, yearning)}
    result = []
    
    for single_photo_row in photo:
        current_sum = 0
        for person_in_photo in single_photo_row:
            current_sum += name_yearning_map.get(person_in_photo, 0)
        result.append(current_sum)
        
    return result